onadata/apps/logger/migrations/0061_auto_20200713_0814.py

```diff
@@ -0,0 +1,26 @@
+# pylint: skip-file
+# Generated by Django 2.2.10 on 2020-07-13 12:14
+
+from django.db import migrations
+from onadata.libs.utils.common_tools import get_uuid
+
+
+def generate_uuid_if_missing(apps, schema_editor):
+    """
+    Generate uuids for XForms without them
+    """
+    XForm = apps.get_model('logger', 'XForm')
+
+    for form in XForm.objects.filter(uuid=''):
+        form.uuid = get_uuid()
+        form.save()
```

Comment:
Won't this be expensive when running migrations - even though it might be one time? Can we do this in the XForm's `save` function; check if that form has a uuid and set one if it doesn't exist.
I guess my worry is just the app going through all XForms checking which one doesn't have a uuid

Reply:
May be not check on the XForm's instance save function but where the uuid is required

Reply:
I'm not entirely sure if this query would be expensive to run(_I'll try analyzing the query on a large database and get back to you_). The `XForm` save function has a check to set the UUID when missing through the `DataDictionary` model.

The main reason this was included was because of `MergedXForms` since they're not created through the `DataDictionary` model. But, I've also modified the `MergedXForm` save to set it when missing...

_I'm not sure at the moment about when `MergedXForms` are updated/saved again after creation_

Reply:
Cool. I guess i'm fine if retrieving xforms without uuid won't be expensive

Reply:
The query isn't expensive to run. _The index we currently have on the UUID field speeds up the query quite a lot_

<img width="1205" alt="Screenshot 2020-07-15 at 15 26 42" src="https://user-images.githubusercontent.com/25849009/87544935-e57d6c00-c6af-11ea-9269-52a863bc0e4e.png">

Reply:
Cool